client/components/mma/cancel/cancellationSaves/digipack/DigiSubThankYouOffer.tsx

```diff
@@ -149,42 +142,39 @@ export const DigiSubThankYouOffer = () => {
 		CancellationContext,
 	) as CancellationContextInterface;
 	const productDetail = cancellationContext.productDetail;
-	const discountMonths = getDiscountMonthsForDigisub(productDetail);
-
-	const {
-		data,
-		loadingState,
-	}: {
-		data: DiscountPreviewResponse | null;
-		loadingState: LoadingState;
-	} = useAsyncLoader(
-		() =>
-			fetchWithDefaultParameters('/api/discounts/preview-discount', {
-				method: 'POST',
-				body: JSON.stringify({
-					subscriptionNumber:
-						productDetail.subscription.subscriptionId,
-					discountProductRatePlanId:
-						getDiscountRatePlanId(discountMonths),
-				}),
-			}),
-		JsonResponseHandler,
-	);
 
 	const location = useLocation();
 	const routerState = location.state as CancellationRouterState;
 
-	const [isDiscountLoading, setIsDiscountLoading] = useState<boolean>(false);
+	const [isApplyDiscountLoading, setIsApplyDiscountLoading] =
+		useState<boolean>(false);
 	const [hasDiscountFailed, setHasDiscountFailed] = useState<boolean>(false);
 
-	if (loadingState == LoadingState.IsLoading) {
+	const [isPreviewDiscountLoading, setIsPreviewDiscountLoading] =
```

Comment:
I removed the asyncLoader as it will handle a 400 response as an error and log it as such - maybe that's fine? But given we expect this endpoint to return 400s for some users I thought we shouldn't do that

Reply:
Good point. How do differentiate the difference between the 400s we expect and actual "bad" 400s?

Reply:
Great shout to remove another AsyncLoader!

Reply:
don't think so - slight risk of getting a 404 if the URLs change but any validation errors I don't think we need to log